gc/base/standard/Scavenger.cpp

```diff
@@ -479,14 +485,100 @@ MM_Scavenger::calculateMaxCacheCount(uintptr_t activeMemorySize)
 	return 5 * (activeMemorySize / (_extensions->scavengerScanCacheMaximumSize + _extensions->scavengerScanCacheMinimumSize));
 }
 
+void
+MM_Scavenger::calculateRecommendedWorkingThreads(MM_EnvironmentStandard *env)
+{
+	if (!_extensions->adaptiveThreadigEnabled() || _extensions->isConcurrentScavengerEnabled()) {
```

Comment:
It might be overkill to check if adaptive is enabled both here and in dispatcher!?

It would be interesting to calculate recommended value  (and report through trace point)  even is adaptive is disabled.

But then, in a case  of this calculation misbehaving (for example div by 0), it's good to have ability to disable it. So, let's play safe and leave the check here (and in Dispatcher too).

Reply:
I agree, it could be interesting to calculate the recommended value regardless. But we should keep the ability to skip the routine in case of "anything". 

By leaving the check in Scavenger,  the check in Dispatcher is defiantly an overkill:

`if (_extensions->adaptiveThreadigEnabled() && (task->getRecommendedWorkingThreads() != UDATA_MAX)) `

`getRecommendedWorkingThreads` will always cause condition to fail unless adaptive threading is enabled. Perhaps this can be simplified